### PR TITLE
refactor(module:docs): refactor docs to standalone mode

### PR DESCRIPTION
**Pull request by Laffery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: scripts/site/_site/doc/app/app.config.server.ts
Comment (Laffery) [inline scripts/site/_site/doc/app/app.config.server.ts:1]: generated from `ng new --ssr` and extended from the original `app.server.module.ts` file

File: scripts/site/_site/doc/app/app.routes.ts [hunk 1]
```diff
@@ -1,16 +1,14 @@
 import { Routes } from '@angular/router';
-import { DEMOComponent } from './_demo/demo.component';
 
 import { DEMO_ROUTES } from './router';
 
 export const routes: Routes = [
   { path: '', pathMatch: 'full', redirectTo: '/docs/introduce/en' },
   ...DEMO_ROUTES,
   { path: 'docs', loadChildren: () => import('./docs/index.module').then(m => m.NzDocsModule) },
-  { path: 'demo', component: DEMOComponent },
```
Comment (Laffery) [inline scripts/site/_site/doc/app/app.routes.ts:10]: Will be redirected to `/introduce/zh` when visiting `/demo`, I wonder whether there are developer take advantage of this or not, so removed

**Review by HyperLife1119 — CHANGES_REQUESTED** [review]
Nice job!

File: scripts/site/_site/doc/app/app.config.server.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const antDesignIcons = AllIcons as {
+  [key: string]: IconDefinition;
+};
+
+const icons: IconDefinition[] = Object.keys(antDesignIcons).map(key => antDesignIcons[key]);
+
+const serverConfig: ApplicationConfig = {
+  providers: [
+    provideServerRendering(),
+    importProvidersFrom(NzIconModule.forRoot(icons))
```
[the remark this answers is not in the document]
Comment (Laffery) [reply scripts/site/_site/doc/app/app.config.server.ts:25]: You are RIGHT
我印象中之前支持了 provideNzConfig 之后，我就感觉 icons 这里的用法有点过时了

File: scripts/site/_site/doc/app/app.component.ts [hunk 3]
```diff
@@ -403,7 +435,7 @@ export class AppComponent implements OnInit {
     const hasLanguage = pathname.match(/(en|zh)(\/?)$/);
     if (language === 'zh-cn' && !hasLanguage) {
       this.nzI18nService.setLocale(zh_CN);
-      this.router.navigate(['docs', 'introduce', 'zh']);
+      this.router.navigate(['docs', 'introduce', 'zh']).then();
```
[the remark this answers is not in the document]
Comment (Laffery) [reply scripts/site/_site/doc/app/app.component.ts:438]: IDE warning 提示，不影响代码逻辑
image

**Review by HyperLife1119 — APPROVED** [review]
LGTM. Nice Job!

**Pull request merged**